### PR TITLE
Remove support for PHP 8.0, add support for PHP 8.3

### PR DESCRIPTION
**Pull request by glo71317**
|    Q          |   A
|-------------- | ------
| Documentation | yes/no
| Bugfix        | yes/no
| BC Break      | yes/no
| New Feature   | yes/no
| RFC           | yes/no
| QA            | yes/no

### CODE REVIEW

**Review by gsteel — CHANGES_REQUESTED** [review]
This needs a full revert with the exception of adding 8.3 to the list of compatible PHP versions.

I believe that most of the dependencies here now have 8.3 compatible releases.

If composer still refuses to install dependencies, you can try adding `.laminas-ci.json` to the root of the project with:
```json
{
     "ignore_php_platform_requirements": {
        "8.3": true
     }
}
```

**Comment by glo71317** [comment]
@gsteel Can you suggest about README Linting and Documentation Linting failures?

**Comment by gsteel** [comment]
The README has trailing spaces and the issue with the docs is the header increment jumps from a level 1 or 2 to a level 4 heading (it should probably be a level 2 or 3 heading depending on context).

See the linting errors in CI: https://github.com/laminas/laminas-http/actions/runs/6707197515

**Comment by froschdesign** [comment]
@glo71317 
Change the headline level from 4:

```markdown
> #### Known issue with libcurl prior to 7.30.0
```

to 3:

```markdown
> ### Known issue with libcurl prior to 7.30.0
```

https://github.com/laminas/laminas-http/blob/76de9008f889bc7088f85a41d0d2b06c2b59c53d/docs/book/client/adapters.md?plain=1#L244

**Comment by glo71317** [comment]
@gsteel @froschdesign I have fixed linting issues, Can you please review the changes and proceed for merging if there is not an issue?

File: composer.json [hunk 1]
```diff
@@ -23,10 +23,10 @@
...
     },
     "require": {
-        "php": "~8.0.0 || ~8.1.0 || ~8.2.0",
-        "laminas/laminas-loader": "^2.8",
+        "php": "~8.1.0 || ~8.2.0 || ~8.3.0",
+        "laminas/laminas-loader": "^2.10",
         "laminas/laminas-stdlib": "^3.6",
-        "laminas/laminas-uri": "^2.9.1",
+        "laminas/laminas-uri": "^2.11",
         "laminas/laminas-validator": "^2.15"
```
Comment (gsteel) [inline composer.json:30]: Would prefer not to bump direct dependencies without a specific reason here.

**Review by gsteel — APPROVED** [review]
Thanks @glo71317

**Pull request merged**